From e3339ad63c0a51e353f714743b8f0124ec2d89ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Max=20G=C3=B6rner?=
 <5477952+MaxG87@users.noreply.github.com>
Date: Mon, 27 Sep 2021 12:24:19 +0200
Subject: [PATCH] Disable IPX for Linux kernel >= 5.15

Commit 6c9b40844751ea30c72f7a2f92f4d704bc6b2927 to the Linux kernel
removed the IPX infrastructure. Commit
ff63261978ee1f00758f447d63276a6f180a81dc repaired the one driver that
relied on that by removing all related function. The changes made there
inspired the fixes here, namely to guard everything by preprocessor
directives.
---
 drivers/net/wireless/rtl8822bs/core/rtw_br_ext.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rtl8822bs/core/rtw_br_ext.c b/drivers/net/wireless/rtl8822bs/core/rtw_br_ext.c
index 9a0effd..d2da04e 100644
--- a/drivers/net/wireless/rtl8822bs/core/rtw_br_ext.c
+++ b/drivers/net/wireless/rtl8822bs/core/rtw_br_ext.c
@@ -14,10 +14,13 @@
  *****************************************************************************/
 #define _RTW_BR_EXT_C_
 
+#include <linux/version.h>
 #ifdef __KERNEL__
 	#include <linux/if_arp.h>
 	#include <net/ip.h>
-	#include <net/ipx.h>
+	#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
+		#include <net/ipx.h>
+	#endif
 	#include <linux/atalk.h>
 	#include <linux/udp.h>
 	#include <linux/if_pppox.h>
@@ -169,6 +172,7 @@ static __inline__ void __nat25_generate_ipv4_network_addr(unsigned char *network
 }
 
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
 static __inline__ void __nat25_generate_ipx_network_addr_with_node(unsigned char *networkAddr,
 		unsigned int *ipxNetAddr, unsigned char *ipxNodeAddr)
 {
@@ -200,6 +204,7 @@ static __inline__ void __nat25_generate_apple_network_addr(unsigned char *networ
 	memcpy(networkAddr + 1, (unsigned char *)network, 2);
 	networkAddr[3] = *node;
 }
+#endif
 
 
 static __inline__ void __nat25_generate_pppoe_network_addr(unsigned char *networkAddr,
@@ -892,6 +897,7 @@ int nat25_db_handle(_adapter *priv, struct sk_buff *skb, int method)
 	/*---------------------------------------------------*/
 	/*         Handle IPX and Apple Talk frame          */
 	/*---------------------------------------------------*/
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
 	else if ((protocol == __constant_htons(ETH_P_IPX)) ||
 		 (protocol == __constant_htons(ETH_P_ATALK)) ||
 		 (protocol == __constant_htons(ETH_P_AARP))) {
@@ -1109,6 +1115,7 @@ int nat25_db_handle(_adapter *priv, struct sk_buff *skb, int method)
 
 		return -1;
 	}
+#endif
 
 	/*---------------------------------------------------*/
 	/*                Handle PPPoE frame                */
